contracts/strategies/auraBAL/Strategy.sol

```diff
@@ -0,0 +1,202 @@
+    /// @dev Can be called by the vault only
+    /// @param _caller - the address calling the harvest on the vault
+    /// @param _minAmountOut -  min amount of LP tokens to receive w/o revert
+    /// @param _lock - whether to lock or swap
+    /// @return harvested - the amount harvested
+    function harvest(
+        address _caller,
+        uint256 _minAmountOut,
+        bool _lock
+    ) public onlyVault returns (uint256 harvested) {
+        // claim rewards
+        auraBalStaking.getReward();
+
+        // process rewards
+        address[] memory _rewardTokens = rewardTokens;
+        for (uint256 i; i < _rewardTokens.length; ++i) {
+            address _tokenHandler = rewardHandlers[_rewardTokens[i]];
+            if (_tokenHandler == address(0)) {
+                continue;
+            }
+            uint256 _tokenBalance = IERC20(_rewardTokens[i]).balanceOf(
+                address(this)
+            );
+            if (_tokenBalance > 0) {
+                IERC20(_rewardTokens[i]).safeTransfer(
+                    _tokenHandler,
+                    _tokenBalance
+                );
+                IRewardHandler(_tokenHandler).sell();
+            }
+        }
+
+        uint256 _wethBalance = IERC20(WETH_TOKEN).balanceOf(address(this));
+        uint256 _balBalance = IERC20(BAL_TOKEN).balanceOf(address(this));
+
+        // Deposit to BLP
+        _depositToBalEthPool(_balBalance, _wethBalance, 0);
+
+        // lock or swap the LP tokens for aura BAL
+        uint256 _auraBalBalance = _lockOrSwap(_lock, _minAmountOut);
+
+        uint256 _stakingAmount = _auraBalBalance;
+
+        if (_auraBalBalance > 0) {
+            // if this is the last call, no fees
+            if (IGenericVault(vault).totalSupply() != 0) {
+                // Deduce and pay out incentive to caller (not needed for final exit)
+                if (IGenericVault(vault).callIncentive() > 0) {
+                    uint256 incentiveAmount = (_auraBalBalance *
+                        IGenericVault(vault).callIncentive()) / FEE_DENOMINATOR;
+                    IERC20(AURABAL_TOKEN).safeTransfer(
+                        _caller,
+                        incentiveAmount
+                    );
+                    _stakingAmount = _stakingAmount - incentiveAmount;
+                }
+                // Deduce and pay platform fee
+                if (IGenericVault(vault).platformFee() > 0) {
+                    uint256 feeAmount = (_auraBalBalance *
+                        IGenericVault(vault).platformFee()) / FEE_DENOMINATOR;
+                    IERC20(AURABAL_TOKEN).safeTransfer(
+                        IGenericVault(vault).platform(),
+                        feeAmount
+                    );
+                    _stakingAmount = _stakingAmount - feeAmount;
+                }
+            }
+            // stake what is left after fees
+            auraBalStaking.stake(_stakingAmount);
```

Comment:
Replace with a call to `stake(_stakingAmount)` perhaps?

Reply:
updated